### PR TITLE
Allow stdin as input for registry apply

### LINKED ISSUE #986: Registry tool: allow input from stdin for `registry apply`
Following the patterns from kubectl (and many other tools), we can use `-` as the identifier for stdin. For example, `cat api.yaml | registry apply --parent projects/my-project/locations/global -f -` should be equivalent to `registry apply --parent projects/my-project/locations/global -f api.yaml`.

### PR DESCRIPTION
**Pull request by seaneganx**
Fixes #986.

### CODE REVIEW

File: cmd/registry/patch/apply.go [hunk 1]
```diff
@@ -44,49 +56,53 @@ func Apply(ctx context.Context, client connection.RegistryClient, path, parent s
...
-			header, items, err := readHeaderWithItems(bytes)
+			return addPatches(client, patches, bytes, fileName, parent)
+		})
+	if err != nil {
+		return err
+	}
+	return patches.run(ctx, jobs)
+}
+
+func addPatches(client connection.RegistryClient, patches *patchGroup, bytes []byte, fileName, parent string) error {
```
Comment (timburks) [inline cmd/registry/patch/apply.go:67]: I think the role of this new function would be clearer if it was a method of patchGroup. Could you make that change and move this below the patchGroup struct declaration?

**Review by timburks — CHANGES_REQUESTED** [review]
Thanks for taking this on!

Mostly thinking aloud here: There's a directory structure described [on the wiki](https://github.com/apigee/registry/wiki/registry-apply#a-recommended-directory-structure-for-exported-registries). `registry export` writes registry contents into this structure and places spec contents in separate files alongside the `info.yaml` files that describe spec metadata. `registry apply` reads these directories and correctly picks up the spec files. That's illustrated in [this directory](https://github.com/apigee/registry/tree/main/cmd/registry/patch/testdata/sample-hierarchical)  and tested [here](https://github.com/apigee/registry/blob/8e09d5c4a35706ef7ea0080779b1efde29979abe/cmd/registry/patch/patch_project_test.go#L43). Since that passes, I think this is probably OK.

Tests here don't seem to be testing inline content (and I think that should be done in the patch package), but I'd like to hold that change anyway pending discussion in #989.

**Comment by timburks** [comment]
@theganyo @seaneganx Is it surprising that there is no code coverage report for this PR? (it was to me, but maybe I'm missing something). I see in the action logs that an error was reported at the end of the compute code coverage step:
```
[2023-01-30T18:29:31.435Z] ['info'] Pinging Codecov: https://codecov.io/upload/v4?package=uploader-0.3.2&token=*******&branch=feat%2Fapply-stdin&build=4046939711&build_url=https%3A%2F%2Fgithub.com%2Fapigee%2Fregistry%2Factions%2Fruns%2F4046939711&commit=9beab1b73b73bd4774069756d6b3b19974c6abc5&job=Go&pr=985&service=github-actions&slug=apigee%2Fregistry&name=&tag=&flags=&parent=
[2023-01-30T18:29:31.815Z] ['error'] There was an error running the uploader: Error uploading to [https://codecov.io:](https://codecov.io/) Error: There was an error fetching the storage URL during POST: 404 - {'detail': ErrorDetail(string='Unable to locate build via Github Actions API. Please upload with the Codecov repository upload token to resolve issue.', code='not_found')}
```

**Comment by theganyo** [comment]
> Is it surprising that there is no code coverage report for this PR?

Maybe https://community.codecov.com/t/upload-issues-unable-to-locate-build-via-github-actions-api/3954 ?

**Review by timburks — APPROVED** [review]
This looks great, thanks!

**Pull request merged**